packages/components/src/components/as-switch/as-switch.tsx

```diff
@@ -0,0 +1,88 @@
+import { Component, Element, Event, EventEmitter, Prop } from '@stencil/core';
+
+@Component({
+  shadow: false,
+  styleUrl: './as-switch.scss',
+  tag: 'as-switch',
+})
+
+export class Switch {
+  /**
+   * HTMLElement wraper for the switch component
+   *
+   * @type {HTMLElement}
+   * @memberof Switch
+   */
+  @Element() public el: HTMLElement;
+
+  /**
+   * Boolean flag to control if the input is checked or not
+   *
+   * @type{boolean}
+   * @memberof Switch
+   */
+  @Prop() public checked: boolean;
+
+  /**
+   * Boolean flag to control when the switch is disabled or not
+   *
+   * @type{boolean}
+   * @memberof Switch
+   */
+  @Prop() public disabled: boolean;
+
+  /**
+   * The input name
+   *
+   * @type{string}
+   * @memberof Switch
+   */
+  @Prop() public name: string;
+
+  /**
+   * The input name
+   *
+   * @type{string}
+   * @memberof Switch
+   */
+  @Event() public change: EventEmitter;
+
+  /**
+   * When the component is attached to the DOM bind the onCLick function
+   */
+  public componentDidLoad() {
+    this.el.addEventListener('click', this._onClick.bind(this));
+  }
+
+  /**
+   * Fire a 'change' event with a boolean parameter if the user clicks on an enabled component
+   */
+  public _onClick() {
+    if (!this.disabled) {
+      this.el.querySelector('input').checked = !this.el.querySelector('input').checked;
```

Comment:
Let's not query DOM three times.
I think we should update checked prop as well, just in case anyone wants to take it from there.